source/common/formatter/substitution_formatter.cc

```diff
@@ -685,310 +685,436 @@ class StreamInfoSslConnectionInfoFieldExtractor : public StreamInfoFormatter::Fi
   FieldExtractor field_extractor_;
 };
 
+// Static lookup table to create specific FieldExtractor
+// based on extractor name.
+// The table is static member of StreamInfoFormatter and is
+// created upon startup.
+absl::flat_hash_map<absl::string_view, StreamInfoFormatter::FieldExtractorCreateFunc>
```

Comment:
this looks good generally, but please use CONSTRUCT_ON_FIRST_USE to avoid a static-init-order fiasco. Thanks!

Reply:
Thanks for reviewing and catching this issue. I changed it to access the hash table via static method which uses CONSTRUCT_ON_FIRST_USE.